File: CinemaNet.Abstractions/Models/Movie/Movie.cs
namespace CinemaNet.Abstractions.Models.Movie;

public class Movie
{
    public string Id { get; init; }

    public string Name { get; init; }

    public int Duration { get; init; }

    public DateTime ReleaseDate { get; init; }

    public string Category { get; init; }

    public string Description { get; init; }
    
    public string PictureUrl { get; init; }
    
    public MovieScreenings[] Screenings { get; set; }
}

File: CinemaNet.Web/src/app/modules/movie/models/movie.ts
import { MovieScreenings } from "./movie-screenings";

export class Movie {
    public id!: string;
    public name!: string;
    public duration!: number;
    public releaseDate!: Date;
    public category!: string;
    public description!: string;
    public pictureUrl!: string;
    public screenings!: MovieScreenings[];
}
